GenerateMacroDefinitionsFile.cmake

```diff
@@ -30,6 +30,10 @@ FILE(APPEND ${GENERATED_G3_DEFINITIONS} "// CMake induced definitions below. See
 
 FOREACH(definition ${G3_DEFINITIONS} )
    FILE(APPEND ${GENERATED_G3_DEFINITIONS} "#define ${definition}\n")
+   if(definition MATCHES ".* .*")
```

Comment:
This is causing a huge amount of warnings with `-Wmacro-redefined`

```
/Users/kjell.hedstrom/ws/GitHub/g3log/src/g3log/generated_definitions.hpp:14:9: 
warning: 'G3_DYNAMIC_LOGGING' macro redefined [-Wmacro-redefined]
#define G3_DYNAMIC_LOGGING
        ^
<command line>:1:9: note: previous definition is here
#define G3_DYNAMIC_LOGGING 1
        ^
1 warning generated.
```

Please compile with the examples generated as well as the unit tests.